File: apps/api/app/schemas/sensitive_tags.py
"""Sensitive-tag (Person.is_protected_subject) admin shapes.

The flag and its enforcement already existed (entity_resolution_service's
merge guard, network_service's PROTECTED-tier graph gating, alert_service's
sensitive-tag proxy — see Person.is_protected_subject) with no way to set
it; these shapes back the write path and admin listing that were missing.
"""

from pydantic import BaseModel

from app.schemas.common import ClassificationLevel


class SensitiveTagUpdate(BaseModel):
    is_protected_subject: bool
    reason: str


class SensitiveSubjectOut(BaseModel):
    id: str
    full_name: str
    classification: ClassificationLevel
    is_protected_subject: bool

File: packages/shared-types/src/sensitive-tags.ts
/**
 * Sensitive-tag admin shapes (mirrors app/schemas/sensitive_tags.py).
 */
import type { ClassificationLevel } from './common'

export interface SensitiveTagUpdate {
  is_protected_subject: boolean
  reason: string
}

export interface SensitiveSubjectOut {
  id: string
  full_name: string
  classification: ClassificationLevel
  is_protected_subject: boolean
}
